subprojects/resources-gcs/src/integTest/groovy/org/gradle/integtests/resource/gcs/fixtures/GcsServer.groovy

```diff
@@ -0,0 +1,497 @@
+                        'Date'            : DATE_HEADER,
+                        'ETag'            : { calculateEtag(file) },
+                        'Server'          : SERVER_AMAZON_S3,
+                        'Accept-Ranges'   : 'bytes',
+                        'Content-Type'    : 'application/octet-stream',
+                        'Content-Length'  : "0",
+                        'Last-Modified'   : RCF_822_DATE_FORMAT.print(new Date().getTime())
+                ]
+            }
+        }
+        expect(httpStub)
+    }
+
+    def stubMetaDataBroken(String url) {
+        stubMetaDataLightWeightGet(url, 500)
+    }
+
+    def stubMetaDataMissing(String url) {
+        stubFileNotFound(url)
+    }
+
+    private stubMetaDataLightWeightGet(String url, int statusCode) {
+        HttpStub httpStub = HttpStub.stubInteraction {
+            request {
+                method = 'GET'
+                path = url
+                headers = [
+                        'Content-Type': "application/x-www-form-urlencoded; charset=utf-8",
+                        'Connection'  : 'Keep-Alive'
+                ]
+            }
+            response {
+                status = statusCode
+                headers = [
+                        'x-amz-id-2'      : X_AMZ_ID_2,
+                        'x-amz-request-id': X_AMZ_REQUEST_ID,
+                        'Date'            : DATE_HEADER,
+                        'Server'          : SERVER_AMAZON_S3,
+                        'Content-Type'    : 'application/xml',
+                ]
+            }
+        }
+       expect(httpStub)
+    }
+
+    def stubGetFile(File file, String url) {
+        HttpStub httpStub = HttpStub.stubInteraction {
+            request {
+                method = 'GET'
+                path = url
+                headers = [
+                        'Content-Type': "application/x-www-form-urlencoded; charset=utf-8",
+                        'Connection'  : 'Keep-Alive'
+                ]
+            }
+            response {
+                status = 200
+                headers = [
+                        'x-amz-id-2'      : X_AMZ_ID_2,
+                        'x-amz-request-id': X_AMZ_REQUEST_ID,
+                        'Date'            : DATE_HEADER,
+                        'ETag'            : { calculateEtag(file) },
+                        'Server'          : SERVER_AMAZON_S3,
+                        'Accept-Ranges'   : 'bytes',
+                        'Content-Type'    : 'application/octet-stream',
+                        'Content-Length'  : { file.length()},
+                        'Last-Modified'   : RCF_822_DATE_FORMAT.print(new Date().getTime())
+                ]
+                body = { file.bytes }
+            }
+        }
+        expect(httpStub)
+    }
+
+    def stubListFile(File file, String bucketName, prefix = 'maven/release/', delimiter = '/') {
+        def xml = new StreamingMarkupBuilder().bind {
```

Comment:
This is probably all mocked somewhere, this is an AWS response type we're parsing